exporters/elasticsearch/src/es_log_exporter.cc

```diff
@@ -0,0 +1,254 @@
+  std::string responseBody = handler->GetResponseBody();
+  std::cout << responseBody << std::endl;
+  if (responseBody.find("\"failed\" : 0") == std::string::npos)
+  {
+    if (options_.console_debug_)
+    {
+      std::cout << "Logs were not written to Elasticsearch correctly, response body:" << std::endl;
+      std::cout << responseBody << std::endl;
+    }
+
+    // TODO: Retry logic
+    return sdklogs::ExportResult::kFailure;
+  }
+
+  return sdklogs::ExportResult::kSuccess;
+}
+
+bool ElasticsearchLogExporter::Shutdown(std::chrono::microseconds timeout) noexcept
+{
+  isShutdown_ = true;
+
+  return true;
+}
+
+json ElasticsearchLogExporter::RecordToJSON(std::unique_ptr<opentelemetry::logs::LogRecord> record)
+{
+  // Create a json object to store the LogRecord information in
+  json log;
+
+  // Write the simple fields
+  log["timestamp"] = record->timestamp.time_since_epoch().count();
+  log["name"]      = record->name;
+  log["body"]      = record->body;
+
+  // Write the severity by converting it from its enum to a string
+  static opentelemetry::nostd::string_view severityStringMap_[25] = {
+      "kInvalid", "kTrace",  "kTrace2", "kTrace3", "kTrace4", "kDebug",  "kDebug2",
+      "kDebug3",  "kDebug4", "kInfo",   "kInfo2",  "kInfo3",  "kInfo4",  "kWarn",
+      "kWarn2",   "kWarn3",  "kWarn4",  "kError",  "kError2", "kError3", "kError4",
+      "kFatal",   "kFatal2", "kFatal3", "kFatal4"};
+  log["severity"] = severityStringMap_[static_cast<int>(record->severity)];
+
+  // Convert the resources into a JSON object
+  json resource;
+  if (record->resource != nullptr)
+  {
+    record->resource->ForEachKeyValue([&](nostd::string_view key,
+                                          common::AttributeValue value) noexcept {
+      resource[key.data()] = ValueToString(value);
+      return true;
+    });
+
+    // Push the attributes JSON object into the main JSON under the "resource" key
+    log.push_back({"resource", resource});
+  }
+
+  // Convert the attributes into a JSON object
+  json attributes;
+  if (record->attributes != nullptr)
+  {
+    record->attributes->ForEachKeyValue([&](nostd::string_view key,
+                                            common::AttributeValue value) noexcept {
+      attributes[key.data()] = ValueToString(value);
+      return true;
+    });
+
+    // Push the attributes JSON object into the main JSON under the "attributes" key
+    log.push_back({"attributes", attributes});
+  }
+
+  // Convert traceid, spanid, and traceflags into strings
+  char trace_buf[32];
+  record->trace_id.ToLowerBase16(trace_buf);
+  log["traceid"] = std::string(trace_buf, sizeof(trace_buf));
+
+  char span_buf[16];
+  record->span_id.ToLowerBase16(span_buf);
+  log["spanid"] = std::string(span_buf, sizeof(span_buf));
+
+  char flag_buf[2];
+  record->trace_flags.ToLowerBase16(flag_buf);
+  log["traceflags"] = std::string(flag_buf, sizeof(flag_buf));
+
```

Comment:
Is there any document defining the mapping between opentelementry log fields to elasic ?

Reply:
No I haven't seen any specs for this, but the Elasticsearch data schema is very flexible so I'm so I was keeping the data as close to the Log Data Model as possible. There's a recent issue for the collector about [supporting logs into their Metrics/Traces ES exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1800), so I'll ping them about it there.